libbpf-rs/src/map.rs

```diff
@@ -62,10 +62,38 @@ impl OpenMap {
         util::parse_ret(ret)
     }
 
+    pub fn set_map_flags(&mut self, flags: u32) -> Result<()> {
+        let ret = unsafe { libbpf_sys::bpf_map__set_map_flags(self.ptr, flags) };
+        util::parse_ret(ret)
+    }
+
+    pub fn set_numa_node(&mut self, numa_node: u32) -> Result<()> {
+        let ret = unsafe { libbpf_sys::bpf_map__set_numa_node(self.ptr, numa_node) };
+        util::parse_ret(ret)
+    }
+
     pub fn set_inner_map_fd(&mut self, inner: &Map) {
         unsafe { libbpf_sys::bpf_map__set_inner_map_fd(self.ptr, inner.fd()) };
     }
 
+    pub fn set_map_extra(&mut self, map_extra: u64) -> Result<()> {
+        let ret = unsafe { libbpf_sys::bpf_map__set_map_extra(self.ptr, map_extra) };
+        util::parse_ret(ret)
+    }
+
+    pub fn set_autocreate(&mut self, autocreate: bool) -> Result<()> {
+        let ret = unsafe { libbpf_sys::bpf_map__set_autocreate(self.ptr, autocreate) };
+        util::parse_ret(ret)
+    }
+
+    pub fn set_pin_path<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        let path_c = util::path_to_cstring(path)?;
+        let path_ptr = path_c.as_ptr();
+
+        let ret = unsafe { libbpf_sys::bpf_map__set_pin_path(self.ptr, path_ptr) };
```

Comment:
I am new to the code base, so apologies if this is a stupid question, but what ensures that `path` lives sufficiently long here? Asked differently, what happens if it goes out of scope while `self` continues to live on? To me it seems as if we'd end up with a dangling pointer.

Reply:
Great question. path is strdup-ed internally by libbpf (https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/tools/lib/bpf/libbpf.c#n7893Z)

Reply:
Ah, thanks for the pointer! Makes sense now.